Cargo.toml

```diff
@@ -58,6 +59,7 @@ env_proxy = "0.4"
 flate2 = "1"
 fontdb = { version = "0.21", default-features = false }
 fs_extra = "1.3"
+git2 = "0.19.0"
```

Comment:
This is a somewhat problematic dependency. It tends to have lots of breaking changes before releases, and distro packaging for it is kind of a pain. Doable, and at least at Arch Linux we do it for a lot of projects, but as a hint to projects I'd suggest not if possible. If the *gitoxide* crate(s) can be used instead the pure Rust implementation there is both faster and more ergonomic. It isn't as feature complete as the libgit2 bindings, but if it covers what you need they make more sense to use unless you have a need for the C bindings.

Reply:
That's a very good point. I'll try change the git implementation in favor of `gitoxide`... Worst case scenario i foresee is that not every functionality is available, so I'll evaluate the possibility of jumping the `git2` dependency and call the system git executable directly. I admit not the purest and elegant solution but it may work as good.

Reply:
> and call the system git executable directly

As I understand it that is not a feasible fallback for Typst. Dependency on a system CLI tool would wreak havoc on the sandboxed model (however weak it may already be) and the WASM compatibility. It would quickly get you into a position where some projects would build in some environments but not others, something Typst has done pretty good at minimizing so far.

Reply:
That's true. Turns out `gitoxide` was enough for what was needed.